### PR TITLE
Bug/Authentication group not found

### PR DESCRIPTION
**Pull request by dscottboggs**
This fixes a bug where a user would be a member would not be recognized as authenticated in the following case:

1. The user is a member of a group for which the glob pattern matches the current request path, for which they **are not** allowed access, and
2. The user is also a member of a group for which the glob pattern matches the current request path, for which they **are** allowed access

This is the issue that was preventing most of the route specs from working, however, there is still an unrelated issue that is still preventing some of them from working

### CODE REVIEW

File: scripts/setup-for-manual-testing.sh [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+dppm server group add "name=${DPPM_USER}s group" id=$GROUP_ID
+dppm server add_user "name=$DPPM_USER" groups=$GROUP_ID
+
+if [ $ADMIN = yes ]; then
+  make_admin
+fi
+
+./bin/dppm server run
+
+rm -r data
```
Comment (j8r) [inline scripts/setup-for-manual-testing.sh]: what is this file? It can rather be done as an integration spec

**Review by j8r — DISMISSED** [review]

File: scripts/setup-for-manual-testing.sh (hunk 1 above)
Comment (dscottboggs) [reply scripts/setup-for-manual-testing.sh]: It's a script for quickly setting up a version of the server for manual testing. I was doing some experimenting with curl and jq to try to figure out this and another issue.

Comment (j8r) [reply scripts/setup-for-manual-testing.sh]: Well, why not. Please make it POSIX, and we are good.

File: scripts/setup-for-manual-testing.sh [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
```
Comment (j8r) [inline scripts/setup-for-manual-testing.sh]: miss new line

Comment (dscottboggs) [reply scripts/setup-for-manual-testing.sh]: :roll_eyes: this is what I get for hopping editors

**Review by j8r — APPROVED** [review]

**Pull request merged**